python/ray/_private/services.py

```diff
@@ -1184,10 +1185,18 @@ def start_dashboard(require_dashboard,
         except ImportError:
             warning_message = (
                 "Not all Ray Dashboard dependencies were found. "
-                "In Ray 1.4+, the Ray CLI, autoscaler, and dashboard will "
-                "only be usable via `pip install 'ray[default]'`. Please "
-                "update your install command.")
-            raise ImportError(warning_message)
+                "To use the dashboard please install Ray like so: `pip "
+                "install ray[default]`.")
+            if require_dashboard:
+                raise ImportError(warning_message)
+            else:
+                if log_once("dashboard_failed_import") and not os.getenv(
+                        "RAY_DISABLE_IMPORT_WARNING") == "1":
+                    warning_message += " To disable this message set " \
+                                       "RAY_DISABLE_IMPORT_WARNING " \
+                                       "environment to 1."
+                    logger.info(warning_message)
```

Comment:
would prefer warnings.warn here too

Reply:
I know we're supposed to be using `warnings.warn` here, but I feel the messages are just weird. This is the default output if I change this to `warnings.warn`:
```
/Users/amog/dev/ray/lib/python3.8/site-packages/ray/_private/services.py:1199: UserWarning: Not all Ray Dashboard dependencies were found. To use the dashboard please install Ray like so: `pip install ray[default]`. To disable this message, set RAY_DISABLE_IMPORT_WARNING env var to '1'.
  warnings.warn(warning_message)
```

as opposed to 
```
2021-07-29 14:53:43,181	INFO services.py:1199 -- Not all Ray Dashboard dependencies were found. To use the dashboard please install Ray like so: `pip install ray[default]`. To disable this message, set RAY_DISABLE_IMPORT_WARNING env var to '1'.
```

Reply:
https://docs.python.org/3/library/warnings.html#available-functions you can stacklevel=2?

Reply:
Yeah I did that, but that just makes the warning refer to 1 level up. But we have multiple levels of functions here.

Reply:
UX aside, there's some practical reasons why warnings.warn is appropriate here: https://stackoverflow.com/a/14762106/14961124

Reply:
more specifically, you can catch and filter warnings easily, while logging has a different intent.

Reply:
Got it makes sense. Though this would mean we only want to use `warnings` on the driver side, and not in workers or the raylet, right?

Reply:
yeah, we should actually never raise any logging messages about dependencies on the workers/raylet (it can really clutter the console if we have many actors/raylets)

Reply:
switched this to warnings.